hawkmoth/docstring.py

```diff
@@ -188,15 +188,15 @@ class TypeDocstring(Docstring):
 
```

Comment:
I like the simplicity of the code and the predictability and requirement to be explicit.

I do need to come with terms about documenting typedeffed anonymous structs, which I think is the main use case here. This change basically requires to you document a named struct and the typedef separately. And it requires you to name the struct when it's otherwise unnecessary. And no, the current thing as exemplified in the commit message is not great either.

I also hope it doesn't break a lot of existing documentation people might have.

Reply:
I really wanted us to allow that use case too. Off the top of my head, we could do something like:

Does it work?

```rst
.. c:type:: some_typedef

   .. c:member:: member
```

Maybe this?

```rst
.. c:type:: some_typedef
   
   .. c:struct:: @anonymous_<hash>

      .. c:member:: member
```

Or this one?

```rst
.. c:type:: some_typedef
   
   :c:struct:`anonymous_<hash>`

.. c:struct:: @anonymous_<hash>

   .. c:member:: member
```

Even then there's still the inconsistency with instantiation of multiple objects / variables of a given anonymous type, and that has no good solution I think.

Either way, these require much more extensive rework and this commit at least removes some odd quirks. A net improvement I'd say. That said, we can also hold this one back for now and let it sink a bit longer if you're not ready for it :sweat_smile: